libyul/optimiser/Suite.cpp

```diff
@@ -310,6 +315,10 @@ void OptimiserSuite::validateSequence(string_view _stepAbbreviations)
 			nestingLevel--;
 			assertThrow(nestingLevel >= 0, OptimizerException, "Unbalanced brackets");
 			break;
+		case ':':
+			assertThrow(nestingLevel == 0, OptimizerException, "Cleanup delimiter may only be placed at nesting level zero");
+			assertThrow(++colonDelimiters <=1, OptimizerException, "Too many colon delimiters");
```

Comment:
I'm really not a fan of having things like `++` or `=` in expressions. Can we do it this way like in the other cases?
```suggestion
			++colonDelimiters;
			assertThrow(colonDelimiters <= 1, OptimizerException, "Too many colon delimiters");
```
I think it's much more readable. You don't even have to think whether the value in the condition has already been incremented or not. It's also very easy to see that you're actually changing the value and not just checking a condition).